scripts/release/Dockerfile

```diff
@@ -0,0 +1,44 @@
+# Dockerfile for building nomad binaries
+# that mimics Vagrant environment as far as required
+# for building the scripts and running provision scripts
+
+FROM ubuntu:16.04
```

Comment:
We probably want to base this on `ubuntu:18.04` at this point?

Reply:
I'd like to keep this in parity with our Vagrant setup.  We should update this along with Vagrant, which is out of scope of this PR.